test/com/google/javascript/jscomp/PeepholeRemoveDeadCodeTest.java

```diff
@@ -732,6 +740,17 @@ public void testFoldAssign() {
     testSame("x.a=x.a");
     test("var y=(x=x)", "var y=x");
     test("y=1 + (x=x)", "y=1 + x");
+
+    // GitHub issue #1745: https://github.com/google/closure-compiler/issues/1745
+    test("Object.create({}).prop = 1", "");
+    test("Object.create(proto).foo.bar.baz = 1", "");
+
+    // add() is side-effect free, but addClass() is not
+    test("var jQuery = new $; jQuery.add().add().bar = 1", "var jQuery = new $;");
+    test("var jQuery = new $; jQuery.add().bar.add().baz = 1", "var jQuery = new $;");
+    test("var jQuery = new $; jQuery.bar.add().baz = 1", "var jQuery = new $;");
```

Comment:
This doesn't seem right to me. Even if the add method is a no-op, this could be adding a new baz property to jQuery.bar, so we shouldn't be removing the assignment.

AFAICT, we can only remove the assignment if the leftmost receiver is an object created by a non-side-effect call. If it's a pre-existing object, we can't remove the assignment.

Or am I missing something?